src/validation.cpp

```diff
@@ -2541,45 +2541,20 @@ static int64_t nTimeFlush = 0;
 static int64_t nTimeChainState = 0;
 static int64_t nTimePostConnect = 0;
 
+//! A single block and block index pointer.
 struct PerBlockConnectTrace {
     CBlockIndex* pindex = nullptr;
     std::shared_ptr<const CBlock> pblock;
-    PerBlockConnectTrace() {}
+    PerBlockConnectTrace(CBlockIndex* index, std::shared_ptr<const CBlock> block) : pindex(index), pblock(std::move(block)) {}
 };
+
 /**
- * Used to track blocks whose transactions were applied to the UTXO state as a
- * part of a single ActivateBestChainStep call.
- *
- * This class is single-use, once you call GetBlocksConnected() you have to throw
- * it away and make a new one.
+ * Used to track blocks that were connected as part of a single
+ * ActivateBestChainStep call. The BlockConnected validationinterface signal is
+ * fired after a single ABCS call for all blocks that were connected in that
+ * ABCS call.
```

Comment:
Nit (feel free to ignore): How would you feel about rewording this to make it really obvious that there's one BlockConnected signal per block? For example:

    Used to track blocks that were connected as part of a single
    ActivateBestChainStep call. After the ABCS call, a separate BlockConnected
    validationinterface signal is fired for each block that was connected
    in the ABCS call.

Reply:
Sure. That seems better. Updated.